core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/StackVerticalScaleV4Request.java

```diff
@@ -27,6 +29,16 @@ public class StackVerticalScaleV4Request implements JsonEntity {
 
     private Long stackId;
 
+    private List<String> hostTemplateRoleGroupName;
```

Comment:
- Missing `@ApiModelProperty`.
- May it be `null`?
- The name should be in plural.
- What is the exact purpose of this list? How does it compare to `group`?

Reply:
In connection with my previous comment: Is `hostTemplateRoleGroupName` supposed to be ever specified by the customer (either directly or via CLI or UI interaction)? If not, and it is to be populated solely from our flow logic, it does not seem to be a good idea to keep this extra field in `StackVerticalScaleV4Request`.

Furthermore, passing down `StackVerticalScaleV4Request` unchanged from the controller to the flow logic is not a good practice. The preferred approach is to have some intermediate DTO type that is converted from the API request / model class and then passed over to any internal calls originating from the controller layer.

Reply:
StackVerticalScaleV4Request is part of the internal Event CoreVerticalScalingTriggerEvent. hostTemplateRoleGroupName is internally populated using flow logic, which contains the group component names from the blueprint for the group. This is being used in multiple places on the flow steps and each flow step has a different event associated with it. Instead of creating this same field multiple times/ calculating the group component names multiple times, I found that the best place will be on the stackverticalscalev4request and it will be cascaded to all steps.

Reply:
Okay, but `StackVerticalScaleV4Request` is part of the CB pubic API (well, public from the UI's point of view anyway; CLI thunderhead services also depend on it). It shall only contain what is relevant for the caller invoking the specific CB API operation, and hence `hostTemplateRoleGroupName` does not belong here.

On the direct usage of `StackVerticalScaleV4Request` inside `CoreVerticalScalingTriggerEvent`, I still think it is a bad practice, likely an oversight of @doktoric when he implemented the thing 8 months ago. :smile: Were it replaced with an internal DTO type, I would be happy to leave `hostTemplateRoleGroupName` part of the DTO (and let it be passed around between the different flow steps as you mentioned, instead of wrapping it in the various trigger & result events).

Reply:
+1 for @lajosrodek suggestion

Reply:
done